calico_upgrade/pkg/calicoupgrade.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/docopt/docopt-go"
+	"github.com/projectcalico/calico/calico_upgrade/pkg/commands"
+)
+
+func main() {
+	doc := `Usage:
+  calico-upgrade [options] <command> [<args>...]
+
+    validate  Validate the v1 formatted data, check that it can be migrated to
+              the v3 format, and output a full report of any migrated names,
+              migration errors, or migrated name conflicts.
+    start     Start the upgrade process.  This will validate the data, pause
+              Calico networking, and migrate the data from v1 to v3 format.
+              Once the data is migrated successfully, the calico/node
+              instances can be upgraded to v3.x.
```

Comment:
should we maybe add a line here saying, new pods cannot be scheduled until the upgrade is finished after this point?